### PR TITLE
Patch/credentials refactor

### PR DESCRIPTION
**Pull request by soorin1990**

### CODE REVIEW

File: Runtime/McSubscriber.cs [hunk 1]
```diff
@@ -329,13 +329,13 @@ private void Reset()
...
-        private bool CheckValidCredentials(SubscribeCredentials credentials)
+        private bool CheckValidCredentials(McCredentials credentials)
         {
             if (credentials == null || string.IsNullOrEmpty(streamName)) return false;
 
             return streamName.Length > 0 &&
-                   credentials.url.Length > 0 &&
-                   credentials.token.Length > 0;
+                   credentials.subscribe_url.Length > 0 &&
+                   credentials.subscribe_token.Length > 0;
```
Comment (Yousif-CS) [inline Runtime/McSubscriber.cs:338]: subscriber token can be null or empty, not like the publisher token

File: Samples~/Credentials.asset [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  m_PrefabAsset: {fileID: 0}
+  m_GameObject: {fileID: 0}
+  m_Enabled: 1
+  m_EditorHideFlags: 0
+  m_Script: {fileID: 11500000, guid: df50d1b2624c9de449be1e2702775130, type: 3}
+  m_Name: Credentials
+  m_EditorClassIdentifier: 
+  accountId: BXKNJs
+  subscribe_url: https://director.millicast.com/api/director/subscribe
+  subscribe_token: 84d9602165583c55bd4e98747fc9fa6a8ea32436ec124c80f66222e94863770e
```
Comment (Yousif-CS) [inline Samples~/Credentials.asset:17]: please remove the tokens from the assets

File: Samples~/Credentials.asset [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  publish_url: https://director.millicast.com/api/director/publish
+  publish_token: 2fdcd054c32632891f524bacd41911cb12a83f525f0128866548d7299ae22bc3
```
Comment (Yousif-CS) [inline Samples~/Credentials.asset:19]: same here

**Review by Yousif-CS — CHANGES_REQUESTED** [review]
Some change requests

**Comment by Yousif-CS** [comment]
can we also change the following: 

`CheckValidCredentials in McSubscriber` to: 
```
        private bool CheckValidCredentials(McCredentials credentials)
        {
            if (credentials == null) return false;

            return !string.IsNullOrEmpty(streamName) &&
                   !string.IsNullOrEmpty(credentials.subscribe_url) &&
                   !string.IsNullOrEmpty(credentials.accountId);
        }
```

and `CheckValidCredentials in McPublisher` to: 
```
    private bool CheckValidCredentials(McCredentials credentials)
    {
      if (credentials == null) return false;

      return !string.IsNullOrEmpty(streamName) &&
             !string.IsNullOrEmpty(credentials.publish_token) &&
             !string.IsNullOrEmpty(credentials.publish_url) &&
             !string.IsNullOrEmpty(credentials.accountId);
    }
```
We also now require the account ID to be provided to the publisher as is needed for the viewer url

File: Samples~/Credentials.asset (hunk 3 above)
Comment (soorin1990) [reply Samples~/Credentials.asset]: done

File: Samples~/Credentials.asset (hunk 2 above)
Comment (soorin1990) [reply Samples~/Credentials.asset]: done

File: Runtime/McSubscriber.cs (hunk 1 above)
Comment (soorin1990) [reply Runtime/McSubscriber.cs]: done

**Review by Yousif-CS — APPROVED** [review]

**Review by Yousif-CS — APPROVED** [review]

**Pull request merged**